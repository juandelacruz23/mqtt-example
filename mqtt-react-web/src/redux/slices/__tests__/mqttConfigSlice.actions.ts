import {
  changeValue,
  connectClient,
  subscribe,
  disconnectClient,
  unsubscribe,
} from "../mqttConfigSlice";

describe("change value action", () => {
  it("should return CHANGE_VALUE with payload", () => {
    const fakePath = "fake path";
    const firstAction = changeValue({ path: fakePath });
    expect(firstAction).toEqual({
      type: "config/CHANGE_VALUE",
      payload: {
        path: fakePath,
      },
    });
    expect(firstAction).toMatchSnapshot();

    const fakeTopic = "fakeTopic";
    const secondAction = changeValue({ topic: fakeTopic });
    expect(secondAction).toEqual({
      type: "config/CHANGE_VALUE",
      payload: {
        topic: fakeTopic,
      },
    });
    expect(secondAction).toMatchSnapshot();
  });
});

describe("connect client action", () => {
  const action = connectClient({
    clientId: "fake id",
    host: "fake host",
    path: "path",
    port: 1234,
  });
  expect(action).toMatchSnapshot();
});

describe("disconnect client action", () => {
  const action = disconnectClient();
  expect(action).toMatchSnapshot();
});

describe("subscribe action", () => {
  const action = subscribe({
    qos: 0,
    topic: "fake topic",
  });
  expect(action).toMatchSnapshot();
});

describe("unsubscribe action", () => {
  const action = unsubscribe("fake topic");
  expect(action).toMatchSnapshot();
});
