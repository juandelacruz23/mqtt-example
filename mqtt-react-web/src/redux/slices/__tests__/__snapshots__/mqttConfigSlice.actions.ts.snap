// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "payload": Object {
    "clientId": "fake id",
    "host": "fake host",
    "path": "path",
    "port": 1234,
  },
  "type": "config/CONNECT",
}
`;

exports[` 2`] = `
Object {
  "payload": undefined,
  "type": "config/DISCONNECT",
}
`;

exports[` 3`] = `
Object {
  "payload": Object {
    "qos": 0,
    "topic": "fake topic",
  },
  "type": "config/SUBSCRIBE",
}
`;

exports[` 4`] = `
Object {
  "payload": "fake topic",
  "type": "config/UNSUBSCRIBE",
}
`;

exports[`change value action should return CHANGE_VALUE with payload 1`] = `
Object {
  "payload": Object {
    "path": "fake path",
  },
  "type": "config/CHANGE_VALUE",
}
`;

exports[`change value action should return CHANGE_VALUE with payload 2`] = `
Object {
  "payload": Object {
    "topic": "fakeTopic",
  },
  "type": "config/CHANGE_VALUE",
}
`;
